watch.ts

```diff
@@ -0,0 +1,48 @@
+import request = require('request');
+import { LineStream } from 'byline';
+import { KubeConfig } from '@kubernetes/typescript-node';
+
+export class Watch {
+    'config': KubeConfig;
+
+    public constructor(config: KubeConfig) {
+        this.config = config;
+    }
+
+    public watch(path: string, queryParams: any, callback: (phase: string, obj: any) => void, done: (err: any) => void): any {
+        let url = this.config.getCurrentCluster().server + path;
+
+        queryParams['watch'] = true;
+        let headerParams: any = {};
+
+        let requestOptions: request.Options = {
+            method: 'GET',
+            qs: queryParams,
+            headers: headerParams,
+            uri: url,
+            useQuerystring: true,
+            json: true
+        };
+        this.config.applyToRequest(requestOptions);
+        
+        let stream = new LineStream();
+        stream.on('data', (data) => {
+            let obj = null;
+            if (data instanceof Buffer) {
+                obj = JSON.parse(data.toString());
+            } else {
+                obj = JSON.parse(data);
+            }
+            callback(obj['type'], obj['object']);
```

Comment:
can we check for existence (i.e. `null` or `undefined`) of `obj['type']` and `obj['object']`

if they don't exist, I believe we can throw from inside the callback

Reply:
done.